src/validation/custom.validation.ts

```diff
@@ -78,11 +78,75 @@ export const checkDateValueIsValid = (errMsg: string, dayStr: string = "", month
 
 export const checkOptionalDate = (dayStr: string = "", monthStr: string = "", yearStr: string = "") => {
   if ( dayStr !== "" || monthStr !== "" || yearStr !== "" ) {
-    checkDateValueIsValid(ErrorMessages.INVALID_DATE, dayStr, monthStr, yearStr);
-    checkDateIsInPastOrToday(ErrorMessages.DATE_NOT_IN_PAST_OR_TODAY, dayStr, monthStr, yearStr);
+    const areDateFieldsPresent = checkDateFieldsPresent(dayStr, monthStr, yearStr);
```

Comment:
I think line 80 is redundant now as it checks if one of the 3 str is present, but then the next line will throw error if any of them are missing. So you can probably remove line 80 and it's closing bracket, as the line 81 checks will overrule them.

Reply:
actually, i can see now that because it is for optional dates, if all fields are missing then it jumps to the return true; - so seems ok, ignore previous comment